tests/test-mkfs

```diff
@@ -52,7 +52,7 @@ echo "Test $testname"
 rm -f "${testname}.out" "${testname}.refimg"
 
 xxd -r "${srcdir}/${testname}.xxd" "${testname}.refimg" || exit 99
-run_mkfs -C -v --invariant $ARGS "${testname}.out" $SIZE || exit 99
+(export SOURCE_DATE_EPOCH=1426325213; run_mkfs -C -v -i 0x1234abcd $ARGS "${testname}.out" $SIZE) || exit 99
```

Comment:
The tests break if using anything but the specific timestamp and volume_id, as given by the ```--invariant``` flag they replace. See here: https://github.com/dosfstools/dosfstools/blob/a8746502cb28ee28a1bbbbcef089b35566103bdc/src/mkfs.fat.c#L1592-L1595.

I didn't investigate further, I simply continued the existing behaviour.

Reply:
So `invariant` now use volume id 0x1234abcd. So update volume id in testing image to match volume id computed from SOURCE_DATE_EPOCH? This could be more cleaner way.

Reply:
I actually like it the way it is; the volume-id is specified as a hex number, so it's easier to spot in the reference image (.xxd file). (SOURCE_DATE_EPOCH is specified in base 10.) And the separate flag provides a separate knob to use in testing (although it is hardcoded in the tests right now.)

Also, I'm not sure @andreasbombe would even like to merge that commit, given [this comment](https://github.com/dosfstools/dosfstools/pull/106#issuecomment-444705187). I'm sort of waiting for confirmation whether I should remove that commit.

I can update the reference images and remove the use of volume-id flag in the test, if that is preferred, but I just thought I'd say what's on my mind first.

Reply:
But this does not test that `SOURCE_DATE_EPOCH` env generates stable volume id.

Reply:
What did you have in mind? To run the test one more time, with different SOURCE_DATE_EPOCH and reference image? I didn't do that because I just wanted to update the test from the special --invariant flag to the generic SOURCE_DATE_EPOCH. I didn't want to make functional changes to the test.

Also, @andreasbombe hasn't replied whether they would like to even deprecate the --invariant flag. So the two last commits might get dropped.

Reply:
> What did you have in mind?

I mean that you are using `SOURCE_DATE_EPOCH` in test, but then volume id is not generated by `SOURCE_DATE_EPOCH`, but overridden by `-i` option.

So it does not test that `SOURCE_DATE_EPOCH` is fully working. What I mean is to use only `SOURCE_DATE_EPOCH` as I wrote in first post.

@andreasbombe has not replied yet, but personally I do not see reason why to not use `SOURCE_DATE_EPOCH` in test if you are introducing it. And I guess that `--invariant` option is there specially only for tests as I do not see any benefit to have static volume id. Static hardcoded volume id just cause problems as it is used as unique identifier.

Reply:
> I mean that you are using `SOURCE_DATE_EPOCH` in test, but then volume id is not generated by `SOURCE_DATE_EPOCH`, but overridden by `-i` option.

Right, volume_id is overridden. But not create_time.

If I remove the -i flag then I have to update the reference image to make the test pass. That means:
* There is already (some) test coverage for SOURCE_DATE_EPOCH.
* Replacing -i with SOURCE_DATE_EPOCH (+ updating ref. image) would mean removing test coverage for the -i code path.

I guess instead of updating the existing tests, we can add new tests with SOURCE_DATE_EPOCH=X and ref. image X, where X=[1, 2]? But personally I feel the coverage is OK as is.

Reply:
We do not have answer from @andreasbombe about removal of `--invariant` option, but I'm for deprecation in next release and later to fully remove it. Therefore I'm fine with removing it from documentation. But also it should be removed from all existing usage, therefore from tests too.

`SOURCE_DATE_EPOCH` seems like a standard solution how to deal with stable timestamps and `--invariant` option is just dosfstools specific.

So I'm for:
* removing `--invariant` option from help and documentation
* updating test scripts and test files to use `SOURCE_DATE_EPOCH`
* do not use `--invariant` in tests anymore
* let `--invariant` to be work in next release with release notes information about deprecation
* removal of `--invariant` option from code after next release